src/RecyclableMemoryStream.cs

```diff
@@ -1087,7 +1087,15 @@ public override void Write(ReadOnlySpan<byte> source)
         /// </summary>
         public override string ToString()
         {
-            return $"Id = {this.Id}, Tag = {this.Tag}, Length = {this.Length:N0} bytes";
+            if (!this.disposed)
+            {
+                return $"Id = {this.Id}, Tag = {this.Tag}, Length = {this.Length:N0} bytes";
+            }
+            else
+            {
+                // Avoid properties because of the dispose check, but the fields themselves are not cleared.
```

Comment:
This is also true of the `length` field. We could add the final length or something to the string.